src/main/ipfs.ts

```diff
@@ -41,25 +41,21 @@ export async function rpcRequest(
   }
 }
 
-export function ipfsKill() {
+export function ipfsTeardown() {
   try {
-    getPidsSync(
-      process.platform === 'win32' ? 'ipfs.exe' : '.synthetix/go-ipfs/ipfs daemon'
-    ).forEach((pid) => {
-      logger.log('Killing ipfs', pid);
-      process.kill(pid);
-    });
-    logger.log('Removing .ipfs/repo.lock');
-    rmSync(path.join(IPFS_PATH, 'repo.lock'), { recursive: true });
-  } catch (_e) {
-    // whatever
+    if (readFileSync(PID_IPFS_FILE_PATH, 'utf8')) {
```

Comment:
no need for `if`. just do `rmSync(PID_IPFS_FILE_PATH, {force: true});` - it will not fail even if file does not exist.
Also only delete PID and repo.lock **after** IPFS was shut down.

Reply:
- removed if()
- removed PID_IPFS_FILE_PATH const
- moved file deletion after the shutdown. BUT, I moved them up for a reason, because the files were not deleted when I interrupted the execution of the program (server) with the keyboard Ctrl+C. **I don't know if this case is important for us or not?**

Reply:
Sounds like node process does not wait for ipfs to fully shutdown? I can see you moved `rm` after shutdown - did it work this time?
I reckon we cannot remove ipfs locks until ipfs if fully off anyway. We could do that with PID.
If there are issues - the order can be 
1. remove pid
2. shutdown ipfs
3. remove locks